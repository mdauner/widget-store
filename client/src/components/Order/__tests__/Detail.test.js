import React from 'react'
import { fireEvent, render, cleanup } from 'react-testing-library'
import 'jest-styled-components'
import { Detail } from '../'

afterEach(cleanup)

describe('DetailOrder', () => {
  test('displays items', async () => {
    const { container } = render(
      <Detail
        id={'1'}
        items={[
          {
            widget: { name: 'Widget Prime' },
            id: '1',
            size: 'XS',
            color: '#F00',
            price: 9.99
          },
          {
            widget: { name: 'Widget Basic' },
            id: '2',
            size: 'XL',
            color: '#00F',
            price: 0.99
          }
        ]}
      />
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  test('calls remove callback on click', async () => {
    const removeItem = jest.fn()
    const { getByTestId } = render(
      <Detail
        items={[
          {
            widget: { name: 'Widget Prime' },
            id: '2',
            size: 'XS',
            color: '#F00',
            price: 9.99
          },
          {
            widget: { name: 'Widget Basic' },
            id: '3',
            size: 'XL',
            color: '#00F',
            price: 0.99
          }
        ]}
        removeItem={removeItem}
      />
    )
    const variantId = '2'

    const removeButton = getByTestId(`removeButton-${2}`)
    fireEvent.click(removeButton)
    expect(removeItem).toHaveBeenCalledWith(variantId)
  })
})
