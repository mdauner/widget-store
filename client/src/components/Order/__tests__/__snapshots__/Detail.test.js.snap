// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailOrder displays items 1`] = `
.c0 {
  background-color: white;
  padding: 1rem;
}

.c1 {
  padding: 1rem 0;
  color: hsl(0,0%,10%);
  border-bottom: 1px solid hsl(0,0%,10%);
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

.c10 {
  padding-top: 1rem;
  font-size: 1.4rem;
  color: hsl(0,0%,10%);
  font-weight: bold;
  border-top: 1px solid hsl(0,0%,10%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 {
  font-size: 1rem;
  color: hsl(0,0%,40%);
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  display: inline-block;
  width: 1rem;
  height: 1rem;
  border-radius: 3px;
  background-color: #F00;
}

.c9 {
  display: inline-block;
  width: 1rem;
  height: 1rem;
  border-radius: 3px;
  background-color: #00F;
}

.c6 {
  font-style: italic;
  margin-right: 0.2rem;
}

.c6:not(:first-of-type) {
  margin-left: 0.3rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  class="c0"
>
  <h3>
    Order 
    1
  </h3>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        Widget Prime
      </div>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          Size:
        </div>
         
        <b>
          XS
        </b>
        <div
          class="c6"
        >
          Color:
        </div>
         
        <div
          class="c7"
          color="#F00"
        />
      </div>
      <div
        class="c8"
      >
        $9.99
      </div>
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        Widget Basic
      </div>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          Size:
        </div>
         
        <b>
          XL
        </b>
        <div
          class="c6"
        >
          Color:
        </div>
         
        <div
          class="c9"
          color="#00F"
        />
      </div>
      <div
        class="c8"
      >
        $0.99
      </div>
    </div>
  </div>
  <div
    class="c10"
  >
    <div>
      Total
    </div>
    <div>
      $10.98
    </div>
  </div>
</div>
`;
